packages/repository/src/__tests__/acceptance/operation-hooks.acceptance.ts

```diff
@@ -0,0 +1,56 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/repository
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {expect} from '@loopback/testlab';
+import {DataSource} from 'loopback-datasource-juggler';
+import {DefaultCrudRepository, juggler} from '../..';
+import {Product, ProductRelations} from '../fixtures/models/product.model';
+
+// This test shows the recommended way how to use @loopback/repository
+// together with existing connectors when building LoopBack applications
+describe('Operation hooks', () => {
+  let repo: ProductRepository;
+  beforeEach(givenProductRepository);
+
+  it('supports operation hooks', async () => {
+    const p = await repo.create({slug: 'pencil'});
+    expect(repo.callCounter).to.equal(2);
```

Comment:
When a check like this fails, it's hard to tell what exactly went wrong, because we don't know which hooks were called and which were not.

Also if somehow the hook registration logic misbehaves and registers the callback for wrong hooks (e.g. `access` instead of `before save`), then we won't catch such problem.

In the past, I have good experience with replacing a counter with an array of hook names called.

```suggestion
    expect(repo.hooksCalled).to.eql(['before save', 'after save']);
```

Reply:
👍

Reply:
Done